Sources/AppStoreConnect/Networking/Response.swift

```diff
@@ -160,11 +106,76 @@ public struct Response<Received: Equatable & Sendable>: Equatable, Sendable {
     /// Unwraps the stored ``URL`` and returns it.
     /// - Returns: URL to a downloaded file on-disk.
     /// - Throws: An error if ``data`` is nil.
-    func decode() throws -> URL where Received == URL {
+    public func decode() throws -> URL where Received == URL {
         guard let data = data else {
-            throw Error.dataAssertionFailed
+            throw ResponseError.dataAssertionFailed
         }
 
         return data
     }
 }
+
+/// Error thrown when the response from the ``Transport`` fails to meet expectations.
+public enum ResponseError: Swift.Error, Equatable {
+    /// The response as requested was unsuccessful, as described by the `statusCode` and `error`.
+    case requestFailure(error: ErrorResponse?, statusCode: StatusCode, response: URLResponse)
+    /// The client has exceeded the rate limit for the current period.
+    case rateLimitExceeded(error: ErrorResponse?, rate: Rate?, response: URLResponse)
+    /// Data was expected to exist or conform to some format but failed.
+    case dataAssertionFailed
+
+    var isRecoverable: Bool {
+        switch self {
+        case .rateLimitExceeded:
+            return true
+        case .requestFailure(_, let statusCode, _) where statusCode == 500:
```

Comment:
i'm pretty sure this is the only acceptable status code (besides rate limit). a good enhancement could be doing the rate limit wait-time automatically, bypassing the retry behavior.